Deploy etcd on all nodes

sgryczan opened the pull request:
* Deploy nodes with AIO configuration
* Update Docker Image to use Terragrunt wrapper
* Remove time_sleep after cert-manager deployment

The `etcdserver` errors described in https://github.com/NetApp/ez-rancher/issues/51 appear to be transient, and related to temporary instability of the `etcd` cluster during the first few minutes after RKE is deployed. 

This updates the Docker image to use a fork of Terragrunt, that has been [modified to detect](https://github.com/sgryczan/terragrunt/blob/master/options/auto_retry_options.go#L20-L21)  these `etcdserver` error strings. This allows us to take advantage of Terragrunt's Auto-Retry feature: https://terragrunt.gruntwork.io/docs/features/auto-retry/

Which allows the deployment succeed on a subsequent attempt.

Example:
```
module.rancher.helm_release.rancher: Still creating... [50s elapsed]
module.rancher.helm_release.rancher: Still creating... [1m0s elapsed]

Error: etcdserver: request timed out

  on ../modules/kubernetes/rancher/main.tf line 143, in resource "helm_release" "rancher":
 143: resource "helm_release" "rancher" {


[terragrunt] 2020/07/28 21:02:02 Encountered an error eligible for retrying. Sleeping 5s before retrying.
[terragrunt] 2020/07/28 21:02:07 Running command: terraform apply -input=false -auto-approve
module.cluster_nodes.tls_private_key.key: Refreshing state... [id=c01ff2e1a61d60179be64132665202fc26c0dfb3]
module.rancher.data.dns_a_record_set.vcenter: Refreshing state...
module.rancher.local_file.ssh_public_key: Refreshing state... [id=e8dec5c982a961c71ed253630ce8819231f1e5f1]
```

Instances of this auto-retry occurring can be found in `Deploy` stage of the following jobs: 
* https://github.com/NetApp/ez-rancher/runs/920809892?check_suite_focus=true
* https://github.com/NetApp/ez-rancher/runs/920686278?check_suite_focus=true

*Note: This will only fix this error for individuals using the docker method to deploy. Individuals deploying directly with TF can simply re-run the deployment to achieve similar results.*

bgentry-ntap commented on Dockerfile:
```diff
@@ -30,12 +30,21 @@ RUN apk add --no-cache --virtual .build-deps curl \
+RUN apk add --no-cache git && \
+  git clone https://github.com/sgryczan/terragrunt.git && \
+  cd terragrunt && \
+  go build -o terragrunt -v && \
```
should we just publish this binary and use it?

sgryczan replied:
Yeah, I think publishing a proper release would probably be the better option here.

sgryczan commented:
I moved converted this to a draft after realizing the resulting image was going to be ~1.6GB due to modifying the base image.

I converted the Dockerfile to use a multi-stage build per https://github.com/NetApp/ez-rancher/pull/62/commits/7a1a161f26b35397d9cd617a9102aa6d5b2fbcc4, but Ben's suggestion https://github.com/NetApp/ez-rancher/pull/62#discussion_r461934972 of downloading a published release might be the better option overall.
